### PR TITLE
Login register page

### PR DESCRIPTION
**Pull request by PawelNizialek**

### CODE REVIEW

File: WebAPI/bookmark-manager.API/appsettings.Development.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "DefaultConnection": ""
+    "DefaultConnection": "Server=bookmark-manager-server.database.windows.net;Database=bookmark-manager;User ID=bookmark-administrator;Password=`E2wB6Au}H#F>%D:;Trusted_Connection=False;Encrypt=True;"
```
Comment (najbardziej) [inline WebAPI/bookmark-manager.API/appsettings.Development.json]: Don't commit it
Now I have to change the password :)

**Comment by najbardziej** [comment]
Task linked: [CU-gz95nd Create login/register page](https://app.clickup.com/t/gz95nd)

File: WebAPI/bookmark-manager.API/appsettings.Development.json (hunk 1 above)
Comment (PawelNizialek) [reply WebAPI/bookmark-manager.API/appsettings.Development.json]: sorry, I forgot about it :)

File: frontend/src/app/components/auth/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import {AuthService} from '../../../services/auth.service';
+
+@Component({
+  selector: 'bm-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  user: User;
+  loginForm = this.fb.group({
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/login/login.component.ts]: I would just create a method for Initialisation
```
loginForm: FormGroup;

createLoginForm(){
  this.loginForm = (...)
}

ngOnInit(){
 this.createLoginForm();
}

```

File: frontend/src/app/components/auth/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    loginFormEmail: ['', [Validators.required, Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$')]],
+    loginFormPassword: ['', [Validators.required, Validators.minLength(6)]],
+  });
+  get loginFormEmail(){
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/login/login.component.ts]: Both methods are redundant

File: frontend/src/app/components/auth/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    return this.loginForm.get('loginFormEmail');
+  }
+  get loginFormPassword(){
+    return this.loginForm.get('loginFormPassword');
+  }
+
+  constructor(private fb: FormBuilder, private authService: AuthService) {
+    this.user = {
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/login/login.component.ts]: It's not necessary

File: frontend/src/app/components/auth/login/login.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      username: '',
+      password: ''
+    };
+  }
+
+  ngOnInit(): void {
+  }
+  loginUser(): void{
+    this.user.username = this.loginForm.get('loginFormEmail')?.value;
+    this.user.password = this.loginForm.get('loginFormPassword')?.value;
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/login/login.component.ts]: You can just simplify that like:
```this.user = Object.assign({}, this.loginFrom.value)```

File: frontend/src/app/components/auth/register/register.component.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { Validators, FormBuilder } from '@angular/forms';
+import { AuthService } from '../../../services/auth.service';
+import { User } from '../../../model/user';
+
+@Component({
+  selector: 'bm-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/register/register.component.ts:11]: Same things like in LoginComponent

File: frontend/src/app/components/auth/register/register.component.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  user: User;
+  registerForm = this.fb.group({
+    registerFormEmail: ['', [Validators.required, Validators.email]],
+    registerFormPassword: ['', [Validators.required, Validators.minLength(6)]],
+    registerFormPasswordConfirmation: ['', Validators.required]
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/register/register.component.ts]: You have to create custom validator for password confirmation
```
passwordMatchValidator(fg: FormGroup){
  return fg.get('registerFormPassword').value === fg.get('registerFormPasswordConfirmation').value ? null : { 'mismatch' : true};
}
```

File: frontend/src/app/components/auth/login/login.component.ts (hunk 4 above)
Comment (PawelNizialek) [reply frontend/src/app/components/auth/login/login.component.ts]: due to Strict Class Initialization property must be assigned in the constuctor

File: frontend/src/app/components/auth/login/login.component.ts (hunk 5 above)
Comment (PawelNizialek) [reply frontend/src/app/components/auth/login/login.component.ts]: ok

File: frontend/src/app/components/auth/register/register.component.ts (hunk 6 above)
Comment (PawelNizialek) [reply frontend/src/app/components/auth/register/register.component.ts:11]: ok

File: frontend/src/app/components/auth/login/login.component.ts (hunk 2 above)
Comment (PawelNizialek) [reply frontend/src/app/components/auth/login/login.component.ts]: same like user property

File: frontend/src/app/components/auth/login/login.component.ts (hunk 3 above)
Comment (PawelNizialek) [reply frontend/src/app/components/auth/login/login.component.ts]: ok

File: frontend/src/app/components/auth/register/register.component.ts (hunk 7 above)
Comment (PawelNizialek) [reply frontend/src/app/components/auth/register/register.component.ts]: ok

File: frontend/src/app/components/auth/login/login.component.ts (hunk 4 above)
Comment (Dangermannn) [reply frontend/src/app/components/auth/login/login.component.ts]: Add ```"strictPropertyInitialization": false,``` into ```"compilerOptions"``` in  _**tsconfig.json**_ . It should help.

File: frontend/src/app/components/auth/login/login.component.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  get loginFormPassword(){
+    return this.loginForm.get('loginFormPassword');
+  }
+
+  constructor(private fb: FormBuilder, private authService: AuthService) {
+    this.user = {
+      username: '',
+      password: ''
+    };
+    this.loginForm = this.fb.group({
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/login/login.component.ts]: Move it to _**ngOnInit()**_.  and remove user initialisation. **Constructor is generally used for DI.**
Moreover you need to change loginFormEmail, loginFormPassword to the names of requested JSON fields' names like: username, password

File: frontend/src/app/components/auth/register/register.component.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import {Validators, FormBuilder, AbstractControl, FormGroup} from '@angular/forms';
+import { AuthService } from '../../../services/auth.service';
+import { User } from '../../../model/user';
+
+function passwordMatcher(c: AbstractControl): { [key: string]: boolean} | null {
```
Comment (Dangermannn) [inline frontend/src/app/components/auth/register/register.component.ts]: I have already given you a short method for that
```
passwordMatchValidator(fg: FormGroup){
  return fg.get('registerFormPassword').value === fg.get('registerFormPasswordConfirmation').value ? null : { 'mismatch' : true};
}
```

```
    this.registerForm = this.fb.group({
      registerFormEmail: ['', [Validators.required, Validators.email]],
        registerFormPassword: ['', [Validators.required, Validators.minLength(6)]],
        registerFormPasswordConfirmation: ['', Validators.required]
    }, {validators: this.passwordMatchValidator}
);
```

File: frontend/src/app/components/auth/login/login.component.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    this.user = Object.assign({}, this.loginForm.value);
+    this.authService.login(this.user)
+      .subscribe(data => {
+        console.log('Successful');
+      });
+  }
+  private createLoginForm(): void{
+    this.loginForm = this.fb.group({
+      username: ['', [Validators.required, Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$')]],
+      password: ['', [Validators.required, Validators.minLength(6)]],
```
Comment (najbardziej) [inline frontend/src/app/components/auth/login/login.component.ts]: backend requires password length = 8, change to match it

**Review by najbardziej — APPROVED** [review]

**Pull request merged**